src/simplify_qdq.cpp

```diff
@@ -274,6 +341,78 @@ struct match_find_quantizable_ops
     }
 };
 
+// Note: scales are not constant b/c of dynamic quantization
+inline auto two_arg_dq(const std::string& scale)
+{
+    return match::name("dequantizelinear")(
+        match::nargs(2), match::arg(1)(match::skip_broadcasts(match::any().bind(scale))));
+}
+
+/**
+ * Handles block quantization for MX types.
+ * Matcher is slightly different because dequantizelinear will not
+ * have zero points.
+ */
+struct match_find_mx_quantizable_ops
+{
+    auto matcher() const
+    {
+        auto dq1 = match::arg(0)(skip_post_dq_ops(two_arg_dq("scale1").bind("dq1")));
+        auto dq2 = match::arg(1)(skip_post_dq_ops(two_arg_dq("scale2").bind("dq2")));
+        return match::name(get_quantizable_op_names())(dq1, dq2);
+    }
+
+    /// Check that quantized input is a supported type
+    bool is_block_supported_type(instruction_ref dq1, instruction_ref dq2) const
+    {
+        // only fp4x2 type currently supported for block quantization
+        const std::set<migraphx::shape::type_t> supported_types = {migraphx::shape::fp4x2_type};
+        auto in1 = skip_dq_inputs_to_pack(dq1->inputs().front());
+        auto in2 = skip_dq_inputs_to_pack(dq2->inputs().front());
+        return (contains(supported_types, in1->get_shape().type()) and
```

Comment:
Why do you need to restrict this to fp4?

Reply:
Implemented matching for block quantization.